src/main/java/com/fasterxml/jackson/databind/ext/KotlinSupport.java

```diff
@@ -0,0 +1,23 @@
+package com.fasterxml.jackson.databind.ext;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+public class KotlinSupport {
+    public static boolean isJvmInlineClassSyntheticConstructor(Constructor<?> ctor) {
+        Class<?>[] params = ctor.getParameterTypes();
+        if (params.length == 0) {
+            return false;
+        }
+
+        Class<?> lastParam = params[params.length - 1];
+        return ctor.isSynthetic() && lastParam.getName().equals("kotlin.jvm.internal.DefaultConstructorMarker");
+    }
```

Comment:
This code also covers constructors that do not actually contain `value class` as an argument.

The following is a class definition in `Kotlin` and its constructors taken from the decompiled result.
You can see that the `value class` is not used, but the argument of the synthetic constructor ends with `DefaultConstructorMarker`.

```kotlin
data class Data(val v: Int = 0)
```

```java
// decompiled

   public Data(int v) {
      this.v = v;
   }

   // $FF: synthetic method
   public Data(int var1, int var2, DefaultConstructorMarker var3) {
      if ((var2 & 1) != 0) {
         var1 = 0;
      }

      this(var1);
   }
```

I can't think of any other way to write it myself, but I think it should at least be supplemented with a comment.